plugins/carbon_black_defense/komand_carbon_black_defense/actions/get_details_for_specific_event/action.py

```diff
@@ -18,42 +14,32 @@ def __init__(self):
             output=GetDetailsForSpecificEventOutput(),
         )
 
-    def run(self, params={}):
-        host = self.connection.host
-        token = self.connection.token
-        connector = self.connection.connector
+    def run(self, params=None):
+
+        if params is None:
+            params = {}
         event_id = params.get(Input.EVENT_ID)
+        id_ = self.connection.get_job_id_for_detail_search(event_id=event_id)
+        self.logger.info(f"The id is: {id_}")
+        if id_ is None:
+            return {Output.EVENTINFO: {}}
+        detail_search_status = self.connection.check_status_of_detail_search(id_)
+
+        # check if status of
+        # detail search is complete by checking if the completed property
+        # in results is not equal to the contacted property
+        for _ in range(0, 9999):
```

Comment:
I FEEL like this could benefit from a timeout param, but maybe that would cause more problems than it would solve?

Reply:
Makes sense...Lmk if the change I made is what you meant

Reply:
Let's move this `t1` timestamp above the `for _ in range(0, 9999)` loop and keep a `datetime.now()` comparison within the loop, such that the loop will break if we don't get a good status response from Cb after 5 seconds. The current implementation will only have a difference of almost 0ms.
We can either hardcode a 5s timeout that way, or introduce an `Input.TIMEOUT` customer-configured parameter as a timeout.

Reply:
@mrinehart-r7 during group review, we were wondering: do you have a preference for which method to go with ^

Reply:
I do like @mberezin-r7 's idea around using timestamps to compare. When @jrose-r7 and I were working on this last Friday we did find that the loop completed very quickly in only about one or two iterations.